src/Motor.Extensions.Hosting.RabbitMQ/RabbitMQMessageHostBuilderExtensions.cs

```diff
@@ -14,11 +17,17 @@ public static class RabbitMQMessageHostBuilderExtensions
         public static void AddRabbitMQWithConfig<T>(this IConsumerBuilder<T> builder, IConfiguration config) where T : notnull
         {
             builder.AddTransient<CloudEventFormatter, JsonEventFormatter>();
-            builder.Configure<RabbitMQConsumerOptions<T>>(config);
-            builder.AddSingleton(sp =>
-                RabbitMQConnectionFactory<T>.From(sp.GetRequiredService<IOptions<RabbitMQConsumerOptions<T>>>().Value));
-            builder.AddConsumer<RabbitMQMessageConsumer<T>>();
-            builder.AddSingleton<IQueueMonitor, RabbitMQQueueMonitor<T>>();
+            var consumerOptions = new RabbitMQConsumerOptions<T>();
+            config.Bind(consumerOptions);
+            var connectionFactory = RabbitMQConnectionFactory<T>.From(consumerOptions);
+            builder.AddConsumer(sp => new RabbitMQMessageConsumer<T>(
+                sp.GetRequiredService<ILogger<RabbitMQMessageConsumer<T>>>(), connectionFactory,
```

Comment:
nit: probably move the `connectionFactory` variable to a new line to have one parameter per line to improve readability?